modules/flippBidAdapter.js

```diff
@@ -12,6 +14,61 @@ const DEFAULT_CURRENCY = 'USD';
 const DEFAULT_CREATIVE_TYPE = 'NativeX';
 const VALID_CREATIVE_TYPES = ['DTX', 'NativeX'];
 
+let userKey = null;
+let cookieSynced = false;
+export const storage = getStorageManager({bidderCode: BIDDER_CODE});
+
+const syncUserKey = (userKey) => {
+  if (!cookieSynced) {
+    triggerPixel(`https://idsync.rlcdn.com/712559.gif?partner_uid=${userKey}`);
+    cookieSynced = true;
+  }
+};
+
+export function getUserKey(options = {}) {
+  if (userKey) {
+    syncUserKey(userKey);
+    return userKey;
+  }
+
+  // If the partner provides the user key use it, otherwise fallback to cookies
+  if ('userKey' in options && options.userKey) {
+    if (isValidUserKey(options.userKey)) {
+      userKey = options.userKey;
+      syncUserKey(userKey);
+      return options.userKey;
+    }
+  }
+
+  const FLIPP_USER_KEY = 'flipp-uid';
+
+  // Grab from Cookie
+  const foundUserKey = storage.cookiesAreEnabled() && storage.getCookie(FLIPP_USER_KEY);
+  if (foundUserKey) {
+    syncUserKey(foundUserKey);
+    return foundUserKey;
+  }
+
+  // Generate if none found
+  userKey = generateUUID();
+
+  if (!userKey) {
```

Comment:
I think it's ok to have empty user key if everything fails, kevel request would still work without a key
Updated the branch to return empty string instead of null anyway

Reply:
Perhaps I haven't described well, I meant in general, do we need this check rather then what's the default value of this condition.
Looks like `generateUUID` always returns string, that's was my question.

Reply:
You're right, just removed the check